app/models/concerns/traverse_data_json.rb

```diff
@@ -0,0 +1,47 @@
+module TraverseDataJson
+  extend ActiveSupport::Concern
+
+  #used to take a hash, deeply search and flatten the results into a file array
+  #block expects an array to be returned
+  def find_file_fields_and_transform(hash, files, &block)
```

Comment:
maybe change name to `find_file_fields_and_transform!` to clarify that this function is modifying an argument (file)

Reply:
I guess, if there is a transformation, it happens inside the block itself.  If the block transforms the underlying data it does, if not it will just return a flattened list.

Reply:
will add ! to this and the hash versions as well.